### DISCUSSION TITLE
Plugin system for umami

### DISCUSSION THREAD

**Discussion by bernhardh** [umami-software/umami#1550] [Ideas]
I know, one of the main benefits of umami is its simplicity and that it only shows data which really matters. But the point is: It differs from project to project, what "really matters". 

Wouldn't it be nice to have something like a plugin system, to extend umami in an easy way, without messing up the standard codebase and basic functionallity? For example, I was looking for displaying a statistic about screen resolutions. I found this: https://github.com/umami-software/umami/issues/469 . Of course, I could try to follow the steps there, add the files, etc. but this would make updates really hard.

**Comment by rohandebsarkar** [comment]
Screen size statistics was added in PR #1304.

**Reply by bernhardh** [reply]
Oh. I now found it ;)

But anyway, the screen resolution was just an example for a potential plugin system.
